drv/auxflash-api/src/lib.rs

```diff
@@ -171,6 +171,55 @@ where
     }
 }
 
+/// Extension functions on the autogenerated `AuxFlash` type
+impl AuxFlash {
+    /// Reads a compressed blob, streaming it to a callback function
+    pub fn get_compressed_blob_streaming<
+        F: Fn(&[u8]) -> Result<(), E>,
+        E: From<AuxFlashError>,
+    >(
+        &self,
+        tag: [u8; 4],
+        f: F,
+    ) -> Result<[u8; 32], E> {
```

Comment:
suuuuper nitpicky (sorry), but this is enough generic bounds that i would strongly recommend putting them in a `where` clause:
```suggestion
    pub fn get_compressed_blob_streaming<F, E>(
        &self,
        [tag: u8; 4],
        f: F,
    ) -> Result<u32, E>
    where 
        F: Fn(&[u8]) -> Result<(), E>,
        E: From<AuxFlashError>,
    {
```

personally, i find that somewhat more readable; it took me a second to figure out this wasn't a function returning a weird tuple or something...

Reply:
Reasonable, fixed!